### PR TITLE
Remove Some Explicit Calls to process.exit and Set the exitCode Instead

### PR DESCRIPTION
**Pull request by SudharakaP**
As per the discussion at, https://github.com/jhipster/generator-jhipster/pull/10948#discussion_r358013955 I've removed some invocations of `process.exit` which are safe. 

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
can you review this one plz @mshima ?

**Comment by mshima** [comment]
@pascalgrimaud sure, but this isn't a known bug, so we shouldn't rush for this one.

The reason I can think for 
```
{
         done();
         process.exit(0);
}
```

with `done`: https://github.com/jhipster/generator-jhipster/blob/08f36bbe77a92140516dc4a50401a202c9001f38/cli/utils.js#L184-L190

Is that `done` doesn't terminate the process when errorMsg is empty.

Why? The process got stuck?

**Comment by SudharakaP** [comment]
@mshima : Agreed. I have the same question with that one. :thinking:

One of the reasons I like removing these `process.exit` calls is that they hide the real errors which I think we should fix rather than forcing the process to end with a positive note. :smile:

**Comment by mshima** [comment]
> Why? The process got stuck?

No, `done` behavior was changed after, at https://github.com/jhipster/generator-jhipster/commit/09ecb9e90f3737454aa0dffe270aa2cbf8bed1cf

But logger.error is about to change behavior too.
I will create a new 'fatal' logger function and start from it.

**Comment by mshima** [comment]
@pascalgrimaud is ok to depend on https://www.npmjs.com/package/debug?

It is really useful to get debug output when running tests:
`
DEBUG=jhipster:debug npm test
`

**Comment by mshima** [comment]
Created:
https://github.com/jhipster/generator-jhipster/issues/10963 for tracking process.exit removal
https://github.com/jhipster/generator-jhipster/pull/10961 so we can migrate logger.error uses one by one and remove fatal when and if we manage to migrate everything.

**Comment by pascalgrimaud** [comment]
@mshima : yes we can use 'debug' if needed

File: cli/utils.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ const initHelp = (program, cliName) => {
             logger.info(`Did you mean ${chalk.yellow(suggestion)}?`);
         }
 
-        process.exit(1);
+        process.exitCode = 1;
```
[the remark this answers is not in the document]
Comment (SudharakaP) [reply cli/utils.js]: @mshima : Okay, I've added it back. :smile:

**Comment by pascalgrimaud** [comment]
I'm adding the label "on hold", to not merge it by mistake
We'll wait the next release for that as it needs to be tested seriously to not have impact

**Comment by pascalgrimaud** [comment]
I'm removing 'on hold' label, as the release v6.6.0 is done

**Comment by DanielFran** [comment]
Close/Reopen to launch tests

**Comment by mshima** [comment]
@SudharakaP can you rebase this PR?
Tests are failing, probably because a bad snapshot.

**Comment by SudharakaP** [comment]
@mshima : I've re-based it; let us see how it goes. :thinking:  :thinking:

**Comment by SudharakaP** [comment]
Close/Open for tests.

**Comment by SudharakaP** [comment]
Okay seems everything is passing now. The only failed test is from a previous run. Interestingly github ci doesn't seem to remove previous failed runs upon close/reopen. :smile:

**Pull request merged**